src/utils/fetch.ts

```diff
@@ -0,0 +1,18 @@
+import crossFetch from 'cross-fetch';
+import type { PineconeConfiguration } from '../data';
+
+export const getFetch = (config: PineconeConfiguration) => {
+  if (config.fetchApi) {
+    // User-provided fetch implementation, if any, takes precedence.
+    return config.fetchApi;
+  } else if (global.fetch) {
+    // If a fetch implementation is already present in the global
+    // scope, use that. This should prevent confusing failures in
+    // nextjs projects where @vercel/fetch is mandated and
+    // other implementations are stubbed out.
+    return global.fetch;
+  } else {
+    // Use ponyfill as last resort
+    return crossFetch;
```

Comment:
So just to confirm my understanding: 
1. If the user provides fetch, use that
2. In Vercel, the global.fetch if statement will be true, so we're good in Vercel land (or do they actually provide `config.fetchApi`?)
3. When running locally in a Pinecone example app, `cross-fetch` will be used, so that network requests the client needs to initialize will succeed

Do I have that right?

Reply:
My understanding is that next.js loads `@vercel/fetch` and it will be available as a global without the need to specifically import it. So the next.js scenario would be the second logical branch here.

The ability to pass it through config is just an escape hatch for anyone that has weird/specific needs around the http client configuration that I don't have bandwidth to address at the moment.

Reply:
Roger, thanks!